bin/ocaml_merlin.ml

```diff
@@ -28,14 +28,12 @@ let term =
              debugging purposes only and should not be considered as a stable \
              ouptut.")
   in
-  Common.set_common common ~targets:[];
+  Common.set_common common ~log_file:No_log_file ~targets:[];
   Scheduler.go ~common (fun () ->
-      Dune_engine.File_tree.init ~recognize_jbuilder_projects:true
-        ~ancestor_vcs:None;
-      Dune_rules.Workspace.init ();
-      ( match dump_config with
+      let open Fiber.O in
+      let+ _workspace = Import.Main.scan_workspace common in
```

Comment:
This is going to scan the workspace and load all the dune files. This can be quite expensive in a large workspace. It feels like the `dune ocaml-merlin` command shouldn't do this. WDYT?

Reply:
I agree that `dune ocaml-merlin` should not do that.

I there a more lightweight way to detect the workspace configured for Merlin ?

Reply:
This should work: `(Workspace.workspace ()).merlin_context`

Reply:
This is what was already done: 

https://github.com/ocaml/dune/blob/78707ffa288bc45dc61de05af481e7ec935f586b/src/dune_rules/merlin_server.ml#L51-L54

However without the previous call to `scan_workspace` this seems to always return `default`, and that was the reason the issue was opened in the first place.

Reply:
Was it loading the right workspace file?